File: src/main/java/parking/Car.java
package parking;

import java.time.Duration;
import java.time.LocalDateTime;

public class Car {
    private String id;
    private LocalDateTime enter;
    private LocalDateTime leave;

    public Car(String id, LocalDateTime enter) {
        this.id = id;
        this.enter = enter;
    }

    private LocalDateTime leave(){

        LocalDateTime leave = LocalDateTime.of(2019, 9,17,10,8,0);
        return leave;
    }

    public void setLeave(LocalDateTime leave) {
        if(leave.isAfter(enter))
            this.leave = leave;
    }

    public Long durationMinute() {
        return Duration.between(enter , leave).toMinutes();
    }

    public Long durationHours() {
        return (long)Math.ceil( Duration.between(enter , leave).toMinutes()/60.0 ) ;
    }

}

File: src/main/kotlin/com/Car.kt
package com

import java.time.Duration
import java.time.LocalDateTime

fun main(args: Array<String>) {

    var enter = LocalDateTime.of(2019, 9,17,8,0,0)
    var leave = LocalDateTime.of(2019, 9,17,10,0,0)
    val car = Car("AA-0001", enter)
    car.leave = leave

    println(car.durationMinute())
    println(car.durationHours())

    var leave2 = LocalDateTime.of(2019, 9,17,10,8,0)
    car.leave = leave2

    println(car.durationMinute())
    println(car.durationHours())
}

class Car(var id : String, var enter: LocalDateTime){
    var leave:LocalDateTime? = null
    set(value) {
        if(enter.isBefore(value))
            field = value
    }

    fun durationMinute() = Duration.between(enter, leave).toMinutes()

    fun durationHours() = Math.ceil(Duration.between(enter, leave).toMinutes()/60.0).toLong()
}